Visit Overview (with graph) widget targets correct graph

spacenate opened the pull request:
"Visit Overview (with graph)" and "Visits Over Time" widgets use the same action name when calling `getLastUnitGraphAcrossPlugins`, which can lead to "Visit Overview (with graph)" targeting the wrong graph when clicking on one of the metrics listed under "Report". This not only leads to the scrolling behavior described in #7086, but it also updates the wrong graph!

By adding an optional parameter to the `getEvolutionGraph` method, the actual calling action can be specified. In this case, by passing through the "index" action name, "Visits Overview (with graph)" can be distinguished from "Visits Over Time".

spacenate committed:
Distinguish which action is actually rendering the VisitsSummary evolution graph

tsteur commented:
I just tested it and noticed one problem. Eg when you select a different metric in "Visits Overview (with graph)" it will reload the widget with the headline again, resulting the headline to be shown twice:

![visitssummary](https://cloud.githubusercontent.com/assets/273120/7575528/98b2cf40-f889-11e4-9c8c-dc86efbb929b.png)

That's why it currently calls `getEvolutionGraph` I think as this one does not render the headline again.

The problem is that the index action will render the headline again. It's a bit weird behaviour that we do not replace the previous headline and hope we can fix this in a future version. 

Maybe we need another method in that controller that basically just forwards everything to `getEvolutionGraph`? That would allow you to still use a different graph id I reckon. Not sure if there is a better fix.

spacenate committed:
Alternate method to reload VisitsSummary_WidgetOverviewGraph graph

spacenate commented:
Ah, good catch! I've added another method as you suggested, and think this is a pretty good interim solution. Also added a few comments to help make everything clear.

It seems to replace the headline, https://github.com/piwik/piwik/blob/master/plugins/CoreHome/javascripts/dataTable.js#L262 would need to check for an existing headline and remove it if necessary—I'm new to this project, so not sure if that would run in to any problems. #7822 mentions an  AngularJS layout for pages which would change all this anyway :)

tsteur commented:
Looks good to merge to me. Would you just mind to replace each tab with 4 whitespaces?

spacenate committed:
Convert tabs to spaces

spacenate commented:
Sure thing!

tsteur commented:
Thx for that! Great to see this fixed.
